src/interface.h

```diff
@@ -19,138 +19,178 @@
  * along with freeserf.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#ifndef _INTERFACE_H
-#define _INTERFACE_H
-
-#include "gui.h"
-#include "viewport.h"
-#include "panel.h"
-#include "game-init.h"
-#include "notification.h"
-#include "list.h"
-#include "popup.h"
-#include "random.h"
-
-/* The length between game updates in miliseconds. */
-#define TICK_LENGTH  20
-#define TICKS_PER_SEC  (1000/TICK_LENGTH)
+#ifndef SRC_INTERFACE_H_
+#define SRC_INTERFACE_H_
+
+#include "src/misc.h"
+BEGIN_EXT_C
+  #include "src/random.h"
+  #include "src/map.h"
+  #include "src/player.h"
+  #include "src/building.h"
+END_EXT_C
+#include "src/gui.h"
 
 #define MAX_ROAD_LENGTH  256
 
-static const int map_building_sprite[] = {
-	0, 0xa7, 0xa8, 0xae, 0xa9,
-	0xa3, 0xa4, 0xa5, 0xa6,
-	0xaa, 0xc0, 0xab, 0x9a, 0x9c, 0x9b, 0xbc,
-	0xa2, 0xa0, 0xa1, 0x99, 0x9d, 0x9e, 0x98, 0x9f, 0xb2
+static const unsigned int map_building_sprite[] = {
+  0, 0xa7, 0xa8, 0xae, 0xa9,
+  0xa3, 0xa4, 0xa5, 0xa6,
+  0xaa, 0xc0, 0xab, 0x9a, 0x9c, 0x9b, 0xbc,
+  0xa2, 0xa0, 0xa1, 0x99, 0x9d, 0x9e, 0x98, 0x9f, 0xb2
 };
 
 typedef enum {
-	MAP_CURSOR_TYPE_NONE = 0,
-	MAP_CURSOR_TYPE_FLAG,
-	MAP_CURSOR_TYPE_REMOVABLE_FLAG,
-	MAP_CURSOR_TYPE_BUILDING,
-	MAP_CURSOR_TYPE_PATH,
-	MAP_CURSOR_TYPE_CLEAR_BY_FLAG,
-	MAP_CURSOR_TYPE_CLEAR_BY_PATH,
-	MAP_CURSOR_TYPE_CLEAR
+  MAP_CURSOR_TYPE_NONE = 0,
+  MAP_CURSOR_TYPE_FLAG,
+  MAP_CURSOR_TYPE_REMOVABLE_FLAG,
+  MAP_CURSOR_TYPE_BUILDING,
+  MAP_CURSOR_TYPE_PATH,
+  MAP_CURSOR_TYPE_CLEAR_BY_FLAG,
+  MAP_CURSOR_TYPE_CLEAR_BY_PATH,
+  MAP_CURSOR_TYPE_CLEAR
 } map_cursor_type_t;
 
+typedef enum {
+  CAN_BUILD_NONE = 0,
+  CAN_BUILD_FLAG,
+  CAN_BUILD_MINE,
+  CAN_BUILD_SMALL,
+  CAN_BUILD_LARGE,
+  CAN_BUILD_CASTLE,
+} build_posibility_t;
```

Comment:
This should be `build_possibility_t`. There are some variables with this type that should also be renamed `posibility` -> `possibility`.